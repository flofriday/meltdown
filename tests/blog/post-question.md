---
title: How to ask (me) a technical question.
date: 2022-07-20
description: I love helping people solve their problems, but sometimes it is just an exhausting experience.
---

I love helping people solve their problems, but sometimes it is just an
exhausting experience.

## 1. _You_ need something

When you ask someone a question you want them to take time out of their day to
help you. This means they are doing you a favor, so you should be considered
of their time. Yes there is also paid support, but making the experience of
helping you more pleasant will get you better results either way.

## 2. No hello

Put the question into the first message you send them. Don't just say hello
and wait for them to respond, don't ask if it would be ok if you'd ask them a
question. Just put the question in the first message.

Sure you can be polite but by putting the question into the first message you
will get much faster responses.

[nohello.net](https://nohello.net) is a dedicated website just for this point
with lots more examples and reasons why this sucks.

## 3. Include as much relevant information as possible

Don't just tell me what went wrong, but also what you tried to do, what the
output was (if there is interesting output) and where the relevant code snipped
is.

However sharing too much information is just useless. If you share a screenshot of the
relevant function I might be able to debug it. But I definitely won't open a
zip of the repository. Most likely because when I read your question I'm often on
mobile and I am just not able to. But more importantly because it shows that
you didn't put much effort into your question and that you are not considered
of my time. Therefore I just don't want to invest the time to hunt down a bug
in an unknown codebase which might take quite a bit.

## 4. Consider you might be wrong

When you ask for help, you probably have a wrong understanding of how your
system works. You also probably have an assumption on what happened, and you
should include them in the description. Just be careful that you also include
an objective description on what actually did happen.

<!--
During my time as a tutor I had some crazy question on the levels of:

> My program keeps segfaulting because gamma rays manipulate the binary,
> producing bad code, what to do?

Which, I guess, technically could happened but most likely is nothing of the
sorts but just you messing up memory management in C.
-->

While theses assumption sometimes are correct, and often the wrong ones are easy
to spot out, sometimes they lead the helping person on the wrong track hindering
them to help you.

## 5. Sharing code right

Code ist not just like any other text.

If the messenger service you use has a formatting option for code with syntax
highlighting, please use it.

If it doesn't, share a screenshot of the code (I personally always prefer
screenshots). And most important with code screenshots:
**Include the line numbers in the screenshot**.

If I see that you missed a semicolon on line 23, I want to tell you exactly
that. And no I won't start counting from the first line of the image or retype
the line (which is unintuitive to do so on mobile keyboards).
