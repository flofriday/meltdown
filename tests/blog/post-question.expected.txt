MarkdownTree metadata:{'title': 'How to ask (me) a technical question.', 'date': '2022-07-20', 'description': 'I love helping people solve their problems, but sometimes it is just an exhausting experience.'}
    Paragraph
        TextNode "I love helping people solve their problems, but sometimes it is just an\nexhausting experience."
    HeaderNode size:2
        TextNode "1. "
        EmphNode
            TextNode "You"
        TextNode " need something"
    Paragraph
        TextNode "When you ask someone a question you want them to take time out of their day to\nhelp you. This means they are doing you a favor, so you should be considered\nof their time. Yes there is also paid support, but making the experience of\nhelping you more pleasant will get you better results either way."
    HeaderNode size:2
        TextNode "2. No hello"
    Paragraph
        TextNode "Put the question into the first message you send them. Don't just say hello\nand wait for them to respond, don't ask if it would be ok if you'd ask them a\nquestion. Just put the question in the first message."
    Paragraph
        TextNode "Sure you can be polite but by putting the question into the first message you\nwill get much faster responses."
    Paragraph
        LinkNode url: https://nohello.net
            TextNode "nohello.net"
        TextNode " is a dedicated website just for this point\nwith lots more examples and reasons why this sucks."
    HeaderNode size:2
        TextNode "3. Include as much relevant information as possible"
    Paragraph
        TextNode "Don't just tell me what went wrong, but also what you tried to do, what the\noutput was (if there is interesting output) and where the relevant code snipped\nis."
    Paragraph
        TextNode "However sharing too much information is just useless. If you share a screenshot of the\nrelevant function I might be able to debug it. But I definitely won't open a\nzip of the repository. Most likely because when I read your question I'm often on\nmobile and I am just not able to. But more importantly because it shows that\nyou didn't put much effort into your question and that you are not considered\nof my time. Therefore I just don't want to invest the time to hunt down a bug\nin an unknown codebase which might take quite a bit."
    HeaderNode size:2
        TextNode "4. Consider you might be wrong"
    Paragraph
        TextNode "When you ask for help, you probably have a wrong understanding of how your\nsystem works. You also probably have an assumption on what happened, and you\nshould include them in the description. Just be careful that you also include\nan objective description on what actually did happen."
    Paragraph
        CommentNode "
During my time as a tutor I had some crazy question on the levels of:

> My program keeps segfaulting because gamma rays manipulate the binary,
> producing bad code, what to do?

Which, I guess, technically could happened but most likely is nothing of the
sorts but just you messing up memory management in C.
"
    Paragraph
        TextNode "While theses assumption sometimes are correct, and often the wrong ones are easy\nto spot out, sometimes they lead the helping person on the wrong track hindering\nthem to help you."
    HeaderNode size:2
        TextNode "5. Sharing code right"
    Paragraph
        TextNode "Code ist not just like any other text."
    Paragraph
        TextNode "If the messenger service you use has a formatting option for code with syntax\nhighlighting, please use it."
    Paragraph
        TextNode "If it doesn't, share a screenshot of the code (I personally always prefer\nscreenshots). And most important with code screenshots:\n"
        BoldNode
            TextNode "Include the line numbers in the screenshot"
        TextNode "."
    Paragraph
        TextNode "If I see that you missed a semicolon on line 23, I want to tell you exactly\nthat. And no I won't start counting from the first line of the image or retype\nthe line (which is unintuitive to do so on mobile keyboards).\n"
